eventually/tests/unittests/authentication/testforgetpassword.py

```diff
@@ -0,0 +1,114 @@
+"""
+Forget Password test module
+===========================
+"""
+
+import json
+
+from django.test import TestCase, Client
+
+from authentication.views import CustomUser
+from utils.jwttoken import create_token
+
+VALID_EMAIL_TOKEN = create_token({"user_id":100, 'email': 'john@email.com'}, 60 * 60)
+INVALID_EMAIL_TOKEN = create_token({"user_id":999, 'email': 'john@email.com'}, 60 * 60)
+INVALID_TOKEN = "eyJ1c2VyX2lkIjoxMDAsImVtYWlsIjoiam9obkBlbWFpbC5"
+
+class TestForgetPassword(TestCase):
+    """ Class for Forget Password tests """
+
+    def setUp(self):
+        """ Set up test browser client """
+        self.client = Client()
+
+        user = CustomUser.objects.create(id=100, email="john@email.com")
+        user.set_password = "QwErTy12234"
+        user.save()
+
+
+    def test_post_positive(self):
+        """ Positive test for forget password POST function """
+
+        link = '/api/v1/user/forget_password/'
+
+        email_exist = json.dumps({"email": "john@email.com"})
+
+        expect_200 = self.client.post(link, email_exist, content_type='application/json')
+        self.assertEqual(expect_200.status_code, 200)
+
+
+    def test_post_bademail(self):
+        """ Negative test for forget password POST function """
+
+        link = '/api/v1/user/forget_password/'
+
+        email_not_exist = json.dumps({"email": "doesnotexist@gmail.com"})
+
+        expect_400 = self.client.post(link, email_not_exist, content_type='application/json')
+        self.assertEqual(expect_400.status_code, 400)
+
+
+    def test_post_noemail(self):
+        """ Negative test for forget password POST function """
+
+        link = '/api/v1/user/forget_password/'
+
+        expect_400 = self.client.post(link, {}, content_type='application/json')
+        self.assertEqual(expect_400.status_code, 400)
+
+
+    def test_put_positive(self):
+        """ Positive test for forget password PUT function """
+
+        new_password_valid = json.dumps({"new_password": "QwErTy1234"})
+        link_without_token = '/api/v1/user/forget_password/'
+        link_user_exist = link_without_token + str(VALID_EMAIL_TOKEN)[2:]
```

Comment:
Maybe it is better to mock `handle_token` method and remove this url creating statement.

Reply:
Moved these duplicated statements to constants